ring/ring.go

```diff
@@ -184,6 +184,7 @@ type Ring struct {
 
 	// Cache of shuffle-sharded subrings per identifier. Invalidated when topology changes.
 	// If set to nil, no caching is done (used by tests, and subrings).
+	// Both maps are protected by mtx.
 	shuffledSubringCache             map[subringCacheKey]*Ring
 	shuffledSubringWithLookbackCache map[subringCacheKey]cachedSubringWithLookback
```

Comment:
I'd move this [under the mutex-hat](https://dmitri.shuralyov.com/idiomatic-go/entries/2)

Reply:
Well, it's already under the mutex hat. Everything above these lines is also protected by mtx. Going to remove the comment I just added.

Reply:
Oh okay, then yes, I think removing it makes it less ambiguous (otherwise it feels like the stuff above isn't protected anymore)